### PR TITLE
Add Python exercise

### PR DESCRIPTION
**Pull request by sampart**
# Why is this needed?

Fixes https://github.com/CodeChica/plus-plus/issues/72

## What does this do?

Adds a new Python lab based on the existing Node one but extended with additional explanations and an optional extension task.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -4,9 +4,8 @@ ARG INSTALL_NODE="true"
 ARG NODE_VERSION="lts/*"
 RUN if [ "${INSTALL_NODE}" = "true" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
 
-# [Optional] Uncomment this section to install additional OS packages.
```
Comment (sampart) [inline .devcontainer/Dockerfile:7]: The section wasn't commented out, so this comment was redundant.

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 node_modules
+*.pyc
```
Comment (sampart) [inline .gitignore:2]: Compiled Python bytecode

**Review by xlgmokha — APPROVED** [review]
This is fantastic! I really like the step by step guide with questions and answers. Thank you very much for adding this lab!

**Comment by xlgmokha** [comment]
I’m trying to sort out why the checks aren’t running against this PR and noticed something. Would it be possible to add a GH Action to run the new Python tests?

**Comment by xlgmokha** [comment]
> I’m trying to sort out why the checks aren’t running against this PR and noticed something. Would it be possible to add a GH Action to run the new Python tests?

Never mind. You already have. 😆 Carry on.

**Pull request merged**